dipy/reconst/dti.py

```diff
@@ -1393,7 +1412,8 @@ def wrapped_fit_tensor(
 
 
 @iter_fit_tensor()
-def wls_fit_tensor(design_matrix, data, return_S0_hat=False):
+@warning_for_keywords()
+def wls_fit_tensor(design_matrix, data, *, return_S0_hat=False):
```

Comment:
I think there is a problem with the tests in this function. `args` is being passed to it.

Reply:
yea, that's true. But i am not able to figure out how and why is `args` being passed.

Reply:
I checked the main function as well, it tells me that 3 positional arguments are allowed but 4 are being passed, but when I check the parameters, there are only 3 positional params, i don't know how or why 1 extra `arg` is being passed.